c2_components/src/mfx_c2_decoder_component.cpp

```diff
@@ -1855,6 +1857,9 @@ void MfxC2DecoderComponent::DoWork(std::unique_ptr<C2Work>&& work)
                 if (C2_OK != res) break;
 
                 mfx_sts = DecodeFrame(bs, std::move(frame_out), &flushing, &expect_output);
+		if (MFX_ERR_MORE_DATA == mfx_sts) {
+			ReleaseReadViews(incoming_frame_index.peeku());
```

Comment:
Is this really needed, what errors you monitored without it? 
At least one output can be generated then we can release the input view.
And future more per my understanding this function should not be called directly,  you can check EmptyReadViews()

Reply:
I removed above code.